src/sidebar/services/test/groups-test.js

```diff
@@ -253,6 +253,93 @@ describe('groups', function() {
         });
       });
     });
+
+    it('shows service groups when logged in', () => {
+      fakeSettings.services = [{ groups: ['abc123'] }];
+      const svc = service();
+
+      // Create groups response from server.
+      const groups = [{ name: 'Public', id: '__world__' }, { name: 'ABC', id: 'abc123'}];
+
+      fakeApi.groups.list.returns(Promise.resolve({
+        token: '1234',
+        data: groups,
+      }));
+
+      return svc.load().then(groups => {
+        const onlyGroupABCShown = groups.every(g => g.id === 'abc123');
+        assert.equal(onlyGroupABCShown, true);
+      });
+    });
+
+    it('only shows service groups that exist', () => {
+      fakeSettings.services = [{ groups: ['abc123', 'no_exist'] }];
+      const svc = service();
+
+      // Create groups response from server.
+      const groups = [{ name: 'Public', id: '__world__' }, { name: 'ABC', id: 'abc123'}];
+
+      fakeApi.groups.list.returns(Promise.resolve({
+        token: '1234',
+        data: groups,
+      }));
+
+      return svc.load().then(groups => {
+        const onlyGroupABCShown = groups.every(g => g.id === 'abc123');
+        assert.equal(onlyGroupABCShown, true);
+      });
+    });
+
+    it('shows no groups when none of the service groups exist', () => {
```

Comment:
This test could be implemented by parametrizing the above test. In JS with mocha that typically looks like this:

```js
[{
  paramA: valueA,
  paramB: valueB,
},{
  paramA: valueC,
  paramB: valueD,
}].forEach(({ paramA, paramB }) => {
   it('does something', () => {
     // Test which uses values of `paramA` and `paramB`
   });
});

Reply:
I just made them into one big test but added a description for each.